FinalBattleShip/app/src/main/java/com/example/finalbattleship/BattleFieldFragment.kt

```diff
@@ -0,0 +1,102 @@
+package com.example.finalbattleship
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import com.example.finalbattleship.Adapters.MyfieldcellRecyclerViewAdapter
+import com.example.finalbattleship.viewmodels.BattleFieldViewModel
+
+/**
+ * A fragment representing a list of Items.
+ */
+class BattleFieldFragment (val listner: MyfieldcellRecyclerViewAdapter.FieldAdapterListner): Fragment(),BattleFieldViewModel.BattlefieldListener {
+
+    private var columnCount = 10
+    lateinit var fieldAdapter: MyfieldcellRecyclerViewAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        arguments?.let {
+            columnCount = it.getInt(ARG_COLUMN_COUNT)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_battle_field_list, container, false)
+
+        fieldAdapter= MyfieldcellRecyclerViewAdapter(context!!, listner)
+        if (view is RecyclerView) {
+            with(view) {
+                layoutManager = when {
+                    columnCount <= 1 -> LinearLayoutManager(context)
+                    else -> GridLayoutManager(context, columnCount)
+                }
+                adapter = fieldAdapter
+            }
+        }
+        return view
+    }
+
+    companion object {
+
+        // TODO: Customize parameter argument names
+        const val ARG_COLUMN_COUNT = "column-count"
+
+        // TODO: Customize parameter initialization
+        @JvmStatic
+        fun newInstance(fiedlListner: MyfieldcellRecyclerViewAdapter.FieldAdapterListner) =
+            BattleFieldFragment(fiedlListner)
+    }
+
+    override fun onShipHeat(x: Int, y: Int) {
```

Comment:
```suggestion
    override fun onShipHit(x: Int, y: Int) {
```

Heat is something you can feel in summer under the Sun)